### PR TITLE
Update password policy connector to compatible with IS 7.0.0

### PR DESCRIPTION
**Pull request by sandushi**
## Purpose
- Need to onboard the password policy connector to IS 7.0.0

## Changes from this PR
- Bump the osgi service version ranges to compatible with IS 7.0.0
- Bump the framework and governance versions
- Change the pwd-reset.jsp to honor the brandings and new layout components
- Improve the password reset enforcer and password change handler to honor the rule based password policy expiry as well
- Fix the failed unit test cases
- Move the common logic to utils class

## Related issue
https://github.com/wso2/product-is/issues/21890

### CODE REVIEW

File: component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java [hunk 1]
```diff
@@ -188,4 +206,230 @@ public static String convertWindowsFileTimeToUnixTime(String windowsFileTime) {
         long millisSinceEpoch = (fileTime / HUNDREDS_OF_NANOSECONDS) - WINDOWS_EPOCH_DIFF;
         return String.valueOf(millisSinceEpoch);
     }
+
+    public static boolean isPasswordExpiredForUser(String tenantDomain, double daysDifference,
+                                                   String passwordLastChangedTime, String tenantAwareUsername, UserStoreManager
+                                                           userStoreManager) throws AuthenticationFailedException,
+            org.wso2.carbon.user.core.UserStoreException {
```
Comment (piraveena) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java:213]: add new line

File: component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java [hunk 2]
```diff
@@ -188,4 +206,230 @@ public static String convertWindowsFileTimeToUnixTime(String windowsFileTime) {
...
+        try {
+            String userId = ((AbstractUserStoreManager) userStoreManager).getUserIDFromUserName(tenantAwareUsername);
```
Comment (piraveena) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java:215]: can this userID be null? Let's do a null check

File: component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java [hunk 3]
```diff
@@ -188,4 +206,230 @@ public static String convertWindowsFileTimeToUnixTime(String windowsFileTime) {
...
+            List<PasswordExpiryRule> passwordExpiryRules =
+                    org.wso2.carbon.identity.password.expiry.util.PasswordPolicyUtils
```
Comment (piraveena) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java:217]: shall we move the class to the import section and simplify it as PasswordPolicyUtils..getPasswordExpiryRules(tenantDomain);

File: component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java [hunk 4]
```diff
@@ -188,4 +206,230 @@ public static String convertWindowsFileTimeToUnixTime(String windowsFileTime) {
...
+                            .getPasswordExpiryRules(tenantDomain);
+            boolean skipIfNoApplicableRules =
+                    org.wso2.carbon.identity.password.expiry.util.PasswordPolicyUtils
+                            .isSkipIfNoApplicableRulesEnabled(tenantDomain);
+            // Apply default password expiry policy if no rules given.
+            if (CollectionUtils.isEmpty(passwordExpiryRules)) {
```
Comment (kayathiri4) [inline component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java:223]: Please make sure this is not null at any case

File: component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java (hunk 3 above)
Comment (sandushi) [reply component/authenticator/src/main/java/org/wso2/carbon/identity/policy/password/PasswordPolicyUtils.java:217]: We already have a PasswordPolicyUtils class in this repo itself. Hence the class is added like this

**Comment by sandushi** [comment]
Close the PR because the changes are done by https://github.com/wso2-extensions/identity-outbound-auth-passwordPolicy/pull/61

**Pull request closed without merging**